### DISCUSSION TITLE
Dnn.Platform v9.11.0 Build Error ./build.ps1 --target=ResetDevSite

### DISCUSSION THREAD

**Discussion by azealer** [dnnsoftware/Dnn.Platform#5324] [General]
I've been spending 3 weeks to build v9.10.0, v9.10.2, and v9.11.0, but no luck. The latest release version 9.11.0 has the similar error. It would be really appreciated if you guys help me out.

/Login.ascx.cs;linenumber=1305;type=warning;]Single line comment should begin with a space.
##vso[task.logissue sourcepath=:/dnnsrc/download/Dnn.Platform-9.11.0/DNN Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs;linenumber=1304;type=warning;]Code should not contain trailing whitespace
##vso[task.logissue sourcepath=:/dnnsrc/download/Dnn.Platform-9.11.0/DNN Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs;linenumber=1308;type=warning;]Code should not contain trailing whitespace
##vso[task.logissue sourcepath=:/dnnsrc/download/Dnn.Platform-9.11.0/DNN Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs;linenumber=1339;type=warning;]Code should not contain trailing whitespace
##vso[task.logissue sourcepath=:/dnnsrc/download/Dnn.Platform-9.11.0/DNN Platform/Website/ErrorPage.aspx.cs;linenumber=1;type=warning;]The file header copyright text should match the copyright text from the settings.
##vso[task.logissue sourcepath=:/dnnsrc/download/Dnn.Platform-9.11.0/DNN Platform/Website/Install/Install.aspx.cs;linenumber=400;type=warning;]Single-line comment should be preceded by blank line
##vso[task.logissue sourcepath=:/dnnsrc/download/Dnn.Platform-9.11.0/Dnn.AdminExperience/Dnn.PersonaBar.Extensions/dnn-resource-manager;type=error;]Can't resolve
An error occurred when executing task 'Build'.

----------------------------------------
Teardown
----------------------------------------
git $reset --hard
HEAD is now at ad73d0b 'backup'
                               git $reset HEAD^
Error: MSBuild: Process returned an error (exit code 1).
PS C:\dnnsrc\download\Dnn.Platform-9.11.0>

**Comment by bdukes** [comment]
The `dnn-resource-manager` error with _Can't resolve_ as a message is a known issue. There's some timing issue within the build process where we're seeing that issue inconsistently. For now our workaround is to keep trying again until you get a successful build.

What's your use case for building DNN? Maybe there's a simpler approach than trying to create a site from source.

**Reply by azealer** [reply]
Brain, thanks for your response. I've been trying to remove Authorize/Un-authorize options on Dnn.PersonaBar.Extensions. Can you give me an advice how to get it w/o whole build success?

**Reply by bdukes** [reply]
Are you meaning within the Users section? You want admins to be able to manage users without have the ability to unauthorize them?

**Reply by bdukes** [reply]
At the least, if you're wanting to adjust the behavior of a section of the Persona Bar, I would extract just that one extension and build it on its own, so you're only replacing one piece, not the entire project. I realize that's easier said than done, with the amount of integration in the various build pieces.

**Reply by azealer** [reply]
Yes, I mean the options within Users section. We have our own user authorization module, so we need to disable the options.

**Reply by azealer** [reply]
If I can extract and build the Users section, It would be great. I can't figure it out how to build only the User section. Can you give me the instruction?

**Reply by bdukes** [reply]
The built-in Persona Bar extensions are bundled together, so the simplest approach would be to only build that project (`Dnn.PersonaBar.Extensions`) from Visual Studio in release mode. A more complete approach would be to take that `Dnn.PersonaBar.Extensions` project and remove all of the other extensions.

There are also some add-on hooks in the React component which will let you override only pieces of the extension, but they aren't documented and were designed by folks who aren't part of the team, so you'd need to reverse engineer them to decide if they'll meet your needs or not. (For example, it looks like [specifying a `getUserMenu` prop to the `UserTable` component](https://github.com/dnnsoftware/Dnn.Platform/search?q=getUserMenu) will let you, at least, add your own menu item, but there may also be something in there to override/remove an item).
